### ISSUE TITLE
Comment stripping from files

### ISSUE THREAD

**Issue by katanacrimson** [koto/phar-util#9]
Should look into stripping comments from files before they're stored in the phar.  While I'm not sure how exactly this could be implemented, I do know that Silex, the Symfony2 microframework https://github.com/fabpot/Silex, does this itself.
If used, it should trim down the size of the phar file substantially if the files have a healthy amount of inline documentation; after all, those lines aren't needed since it's being packaged into a phar.

Anways, this seems to be implemented using a method defined in the Symfony kernel, looking at this function: https://github.com/fabpot/Silex/blob/master/src/Silex/Compiler.php#L70
Which, in turn, apparently uses a giant loop over `token_get_all()` 
(documentation: http://us3.php.net/manual/en/function.token-get-all.php)

Consider this a feature request, I guess.

If I get the chance in a few weeks/months, I'll see if I can work out implementation myself but right now I'm pressed for time as I've got several papers to do for english, a speech to hammer out, and exams on the horizon.

**Comment by koto** [comment]
There's a potential incompatibility if files packages in Phar archives use meta-programming to get code configuration from inside comments (it's rarely used, but could be). Still, this would trim down the size like you said. Give it a go if you have time.

**Comment by katanacrimson** [comment]
As noted in the issue reference regarding 65916ac, I'm taking a swing at this now.  Looking at using `php_strip_whitespace()` specifically, as it shouldn't matter how code is formatted.  Feature is set to default to off, as well.

@koto, how's it look to you atm?

Once I test it out and make sure it works, I'll open up a pull request.

Also, apologies for the reference spammy.  Accidentally committed on master.

ED: Relevant branch here: https://github.com/damianb/phar-util/tree/feature-stripcmts

**Comment by katanacrimson** [comment]
Did another commit after I tested it (and found zero effect, no stripping was actually occurring with that first commit!), and I've got a before-and-after on a project of mine where I use phar-build.

Take a look at yukari.phar, the Yukari core (it includes the OpenFlame DBAL and the OpenFlame Framework, plus the submodules for `emberlabs\materia` and `yukari\core`).
Also ignore the chown/chmod, it's an NTFS partition mounted in my linux distro. ;)

```
obsidian@lithion ~/code/yukari/main $ ls -lAh lib/
total 329K
drwxrwxrwx 1 root root 4.0K 2011-10-31 14:19 addons
-rwxrwxrwx 1 root root 314K 2011-11-01 20:30 yukari.phar
-rwxrwxrwx 1 root root  272 2011-11-01 20:35 yukari.phar.pubkey
```

Now, by setting the Yukari makefile to strip files matching the space delimited regexp set of `.php$`, we get this:

```

obsidian@lithion ~/code/yukari/main $ ls -lAh lib/
total 161K
drwxrwxrwx 1 root root 4.0K 2011-10-31 14:19 addons
-rwxrwxrwx 1 root root 153K 2011-11-01 20:35 yukari.phar
-rwxrwxrwx 1 root root  272 2011-11-01 20:35 yukari.phar.pubkey
```

Nearly a 53% savings in filesize.

I've yet to perform a test to see if everything runs fine however, that's next for me to do.  Need to finish my chem homework before I take a whack at that however.
